pkg/cloud/metadata_k8s.go

```diff
@@ -0,0 +1,83 @@
+package cloud
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type KubernetesAPIClient func() (kubernetes.Interface, error)
```

Comment:
nit: What's the reason for the metadata_ prefix for all files?  This seems to just be a Kubernetes client wrapper, maybe client.go or kubernetes.go makes for sense.

Reply:
Otherwise lgtm, will merge if you prefer that name.

Reply:
the main purpose is for  `KubernetesAPIInstanceInfo` to return a Metadata, i.e. how to retrieve Metadata/InstanceInfo from k8s versus from ec2. (and later, one to retieve from ECS for fargate environment). the function you are looking at is just wrapper stuff for testing, KubernetesAPIInstanceInfo is the important one.

Reply:
Yeah that makes sense and I see the reasoning for the similar file names, especially given they all are different mechanisms for finding the same information.